cyto_dl/callbacks/tabular_saver.py

```diff
@@ -0,0 +1,93 @@
+import warnings
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import torch
+from lightning.pytorch.callbacks import Callback
+
+
+class SaveTabularData(Callback):
+    """Callback to save tabular data to disk as a .csv or .parquet after prediction."""
+
+    def __init__(self, save_dir, meta_keys=[], as_parquet: bool = True):
+        """
+        Parameters
+        ----------
+        save_dir: str
+            directory to save the tabular data
+        meta_keys: list
```

Comment:
maybe assert that CellId must be included as metadata?

Reply:
I think I'm missing what you're getting at - so that each row has unique metadata?

Reply:
I'm not sure if I want to enforce that since the metadata could be super different between use cases, e.g. for endos I'm saving out timepoint, starting x and y coordinates for the crop, and the name of the movie it came from